src/test/scala/ie/zalando/fabric/gateway/service/IngressDerivationChainSpec.scala

```diff
@@ -479,6 +480,47 @@ class IngressDerivationChainSpec extends FlatSpec with MockitoSugar with Matcher
     svcRestrictedLimits shouldBe Set("a")
   }
 
+  "Rate limit Routes" should "all have inlineContentIfStatus to ensure they reutnr a json response" in {
+    val gatewayPaths: GatewayPaths = Map(
+      PathMatch("/api/resource") -> PathConfig(
+        Map(
+          Get -> ActionAuthorizations(NEL.of("uid"),
+                                      Some(RateLimitDetails(10, PerMinute, Map("a" -> 20, "b" -> 25, "c" -> 30))),
+                                      InheritedWhitelistDetails,
+                                      UserWhitelist),
+          Post -> ActionAuthorizations(NEL.of("uid"),
+                                       Some(RateLimitDetails(2, PerMinute, Map("a" -> 20))),
+                                       InheritedWhitelistDetails,
+                                       UserWhitelist)
+        ))
+    )
+
+    val gwSpec = GatewaySpec(
+      SchemaDefinedServices(Set(IngressBackend("host", Set(ServiceDescription("svcName"))))),
+      Set.empty[String],
+      WhitelistConfig(Set("a"), Enabled),
+      DisabledCors,
+      EmployeeAccessConfig(ScopedAccess),
+      None,
+      gatewayPaths
+    )
+
+    val rateLimitRoutes = Await
+      .result(
+        ingressDerivationLogic.deriveRoutesFor(gwSpec, GatewayMeta(DnsString.apply("my-test-gw"), "ns", None, Map.empty)),
+        10.seconds
+      )
+      .filter(_.metadata.name.contains("-rl"))
+    val rateLimitRouteFilters = rateLimitRoutes.map(getFilters)
+
+    rateLimitRoutes should not be empty
+    all(rateLimitRouteFilters) should contain(RATE_LIMIT_RESPONSE)
+    rateLimitRouteFilters.foreach { filters =>
+      filters.lastIndexOf(RATE_LIMIT_RESPONSE) should be < filters.indexWhere(
+        _.skipperStringValue().contains("clusterClientRatelimit"))
```

Comment:
This seems a bit brittle, is it safe? Assume that the index of the response is higher than the index of clusterClientRatelimit

Reply:
The filter that sets the json response needs to be before clusterClientRatelimit, since they're evaluated in reverse order for the response.

`inlineContentIfStatus(429, "{}", "application/json") -> clusterClientRatelimit()`

If they are the other way around, then the response will be plaintext since clusterClientRatelimit gets the last word

Reply:
Actually I'll add a comment about that